cow_amm/cow_amm_lp_and_tvl_4047078

```diff
@@ -18,31 +38,70 @@ select day, lp_supply, sum(lp_supply) over (order by day) as total_lp
 from get_lp_balance
 )
 
+-- lp token total supply without date gaps
+ , lp_total_supply as (
+SELECT
+        date_range.day,
+        COALESCE(v.total_lp, LAG(v.total_lp) OVER (ORDER BY date_range.day)) AS total_lp
+    FROM
+        date_range
+    LEFT JOIN
+        total_lp v ON date_range.day = v.day
+)
+-- Compute tvl by multiplying each day's closing price with bet transfers in and out of the pool
 , get_tvl as (
-select 
-  date(time) as day, 
-  time, 
-  blockchain, 
-  cow_amm_address, 
-  token_1_balance_usd+token_2_balance_usd as tvl,
-  RANK() OVER (PARTITION BY date(time) ORDER BY time DESC) AS tvl_rank
-from  dune.cowprotocol.result_balancer_cow_amm_base_query_v_2
-where cow_amm_address={{cow_amm_pool}}
+SELECT x.day, SUM(amount * price_close) as total_tvl FROM (
+    SELECT date(evt_block_time) as day, 
+            symbol, 
+            -(value/POW(10,decimals)) as amount
+    FROM erc20_ethereum.evt_transfer a LEFT JOIN tokens.erc20 b ON a.contract_address = b.contract_address and blockchain in (select blockchain from cow_amm_pool)
+    where "from"  in (select address from cow_amm_pool)
+        and (a.contract_address IN (select token_1_address from cow_amm_pool) or 
+        a.contract_address IN (select token_2_address from cow_amm_pool))
+      
+    UNION ALL
+    SELECT date(evt_block_time) as day, 
+            symbol, 
+            (value/POW(10,decimals)) as amount
+    FROM erc20_ethereum.evt_transfer a LEFT JOIN tokens.erc20 b ON a.contract_address = b.contract_address and blockchain in (select blockchain from cow_amm_pool)
+    where "to"  in (select address from cow_amm_pool)
+        and (a.contract_address IN (select token_1_address from cow_amm_pool) or 
+        a.contract_address IN (select token_2_address from cow_amm_pool))
+) x join prices.usd_daily y
+    ON blockchain in (select blockchain from cow_amm_pool) and x.symbol = y.symbol and y.day=x.day
+group by 1
 )
 
 , total_tvl as (
-select day,  blockchain, cow_amm_address, tvl as total_tvl
+select day, sum(total_tvl) over (order by day) as total_tvl
```

Comment:
What happens here if a token didn't trade for a day? TVL may have gaps.

Reply:
addressed that the same way as for the lp supply